internal/verification/verification.go

```diff
@@ -120,56 +125,67 @@ func (v Validator) VerifyDataIntegrityCredential(ctx context.Context, credential
+	headers, jwt, vp, err := integrity.ParseVerifiablePresentationFromJWT(token.String())
 	if err != nil {
-		return nil, err
-	}
-	var proofMap map[string]any
-	if err = json.Unmarshal(proofBytes, &proofMap); err != nil {
-		return nil, err
+		return errors.Wrap(err, "parsing JWT presentation")
 	}
-	return proofMap[key], nil
-}
 
-func (v Validator) VerifyJWT(ctx context.Context, did string, token keyaccess.JWT) error {
-	// parse headers
-	headers, err := keyaccess.GetJWTHeaders([]byte(token))
-	if err != nil {
-		return sdkutil.LoggingErrorMsg(err, "could not parse JWT headers")
+	// get key to verify the presentation with
+	issuerKID := headers.KeyID()
+	if issuerKID == "" {
+		return errors.Errorf("missing kid in header of presentation<%s>", jwt.JwtID())
 	}
-	jwtKID, ok := headers.Get(jws.KeyIDKey)
-	if !ok {
-		return sdkutil.LoggingNewError("JWT does not contain a kid")
+	issuerDID, err := v.didResolver.Resolve(ctx, jwt.Issuer())
+	if err != nil {
+		return errors.Wrapf(err, "error getting issuer DID<%s> to verify presentation<%s>", jwt.Issuer(), jwt.JwtID())
 	}
-	kid, ok := jwtKID.(string)
-	if !ok {
-		return sdkutil.LoggingNewError("JWT kid is not a string")
+	issuerKey, err := did.GetKeyFromVerificationMethod(issuerDID.Document, issuerKID)
+	if err != nil {
+		return errors.Wrapf(err, "error getting key to verify presentation<%s>", jwt.JwtID())
 	}
 
-	// resolve key material from the DID
-	pubKey, err := didint.ResolveKeyForDID(ctx, v.didResolver, did, kid)
+	// construct a verifier and verify the signature on the presentation
+	// note: this also verifies the signature of each credential in the presentation
+	verifier, err := jwx.NewJWXVerifier(issuerDID.ID, issuerKID, issuerKey)
 	if err != nil {
-		return sdkutil.LoggingError(err)
+		return errors.Wrapf(err, "error constructing verifier for presentation<%s>", jwt.JwtID())
 	}
-
-	// construct a signature validator from the verification information
-	verifier, err := keyaccess.NewJWKKeyAccessVerifier(did, kid, pubKey)
+	_, _, _, err = integrity.VerifyVerifiablePresentationJWT(ctx, *verifier, v.didResolver, token.String())
 	if err != nil {
-		return sdkutil.LoggingErrorMsgf(err, "could not create validator for kid %s", kid)
+		return errors.Wrapf(err, "error verifying presentation<%s>", jwt.JwtID())
 	}
 
-	// verify the signature on the credential
-	if err = verifier.Verify(token); err != nil {
-		return sdkutil.LoggingErrorMsg(err, "could not verify the JWT signature")
+	// for each credential in the presentation, run a set of static verification checks
+	for _, cred := range vp.VerifiableCredential {
+		_, _, normalizedCred, err := parsing.ToCredential(cred)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing credential in presentation<%v>", cred)
+		}
+		if err = v.staticValidationChecks(ctx, *normalizedCred); err != nil {
+			return errors.Wrapf(err, "error running static validation checks on credential in presentation<%v>", normalizedCred.ID)
+		}
```

Comment:
Consider reusing some of the stuff we have:
```go
	creds, err := NewCredentialContainerFromArray(vp.VerifiableCredential)
	if err != nil {
		return err
	}
	for _, cred := range creds {
		if err := v.Verify(ctx, cred); err != nil {
			return err
		}
	}
```

Reply:
nice forgot about this. ideally some of this logic should move to the sdk to mirror the credential functionality